sql/src/main/java/io/crate/analyze/TableIdentsExtractor.java

```diff
@@ -0,0 +1,204 @@
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.collect.Lists;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.AnalyzedRelationVisitor;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.analyze.relations.QueriedDocTable;
+import io.crate.analyze.relations.TableFunctionRelation;
+import io.crate.analyze.relations.TableRelation;
+import io.crate.analyze.symbol.Aggregation;
+import io.crate.analyze.symbol.DynamicReference;
+import io.crate.analyze.symbol.FetchReference;
+import io.crate.analyze.symbol.Field;
+import io.crate.analyze.symbol.Function;
+import io.crate.analyze.symbol.InputColumn;
+import io.crate.analyze.symbol.Literal;
+import io.crate.analyze.symbol.MatchPredicate;
+import io.crate.analyze.symbol.ParameterSymbol;
+import io.crate.analyze.symbol.SelectSymbol;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.analyze.symbol.SymbolVisitor;
+import io.crate.analyze.symbol.Value;
+import io.crate.metadata.Reference;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.TableIdent;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+public class TableIdentsExtractor {
+
+    private static final RelationExtractor RELATION_EXTRACTOR = new RelationExtractor();
+    private static final SymbolExtractor SYMBOL_EXTRACTOR = new SymbolExtractor();
+
+    public static Collection<TableIdent> extract(Collection<? extends Symbol> symbols) {
+        Collection<TableIdent> tableIdents = new ArrayList<>(symbols.size());
+        for (Symbol symbol : symbols) {
+            Collection<TableIdent> tableIdentList = SYMBOL_EXTRACTOR.process(symbol, null);
+            if (tableIdentList != null) {
+                tableIdents.addAll(tableIdentList);
+            }
+        }
+        return tableIdents;
+    }
+
+    private static class SymbolExtractor extends SymbolVisitor<Void, Collection<TableIdent>> {
```

Comment:
hm, the name sounds like it extracts symbols. what about something like TableIdentSymbolVisitor and the one below TableIdentRelationVisitor